.clippy.toml

```diff
@@ -1 +1,12 @@
 msrv = "1.50"
+disallowed-types = [
```

Comment:
Are we only using HashMap in state-res in next? Either way I think it should have its own clippy config without these disallowed types.